### DISCUSSION TITLE
Xdrip connection problem with Miaomiao 3

### DISCUSSION THREAD

**Discussion by KimF2020** [NightscoutFoundation/xDrip#2263] [Q&A]
Hello everyone,

I got a big problem with Miaomiao (2 and 3) and Xdrip. Everything worked fine for a while. And now it's not working anymore and that's a big problem for my diabetes. I cannot live without the alarms because i cannot feel it well when i'm high or low. 

So i bought the Miao 3 because i thought the 2 was broken.  Now i got the 3 and i got the same problem. First the Xdrip cannot find the Miaomiao, after a while and reseting the Miaomiao, it can be find. But then the problems start. It's connecting but then after a minute or so, it's disconnecting. After i clicked on 'Start collector again', it's connecting but then the same problem... the connection failed again. I tried everything what i can find on Github en Gitter. I uninstalled Xdrip and downloaded the newest version van 3th of august 2022. I have emptied the cache several times in the Bluetooth settings and the Xdrip settings. All the settings in Xdrip are correct (i set all the settings conform what i can find in Github en Gitter). I restart my phone many many times.  But still it's connectiong and shortly after it's disconnecting. 

I don't know whats wrong. The only thing i see is that the battery of the Miaomiao 3 is going empty/low very fast while it have to last for 14 days. 

My phone is Android, Galaxy Z Flip3 5G, model: SM-F711B. FYI: on this phone Xdrip with the Miaomiao 2 worked complety fine for more than 5 months. 

I hope somebody got the answer for me of can help me. I am desperate. Thank you for helping me.

**Comment by Navid200** [comment]
@psonnera

**Comment by psonnera** [comment]
@KimF2020 try to start with these settings. Exactly as the pictures show them. https://xdrip.readthedocs.io/en/latest/install/libreBT/#bridge-settings and can you confirm it's a Libre 1? As for these bridges, only the first one had a good battery IMHO. Make sure you always disconnect it when it's charged. Another point you should check is the bluetooth system app settings (it's usually hidden in the apps list, try to make it visible using the three dots menu up right, depends on phones).

**Reply by KimF2020** [reply]
@psonnera thank you for your reply.  I've got al Libre 2. All the settings are correct as shown in the docs. It's really weird suddenly later in the evening the connection starts working and for the complete nigth it works! I don't know why. The only thing what is a problem for know is that the Miaomiao 3 is already empty while i charged for the 100% yesterday.

**Reply by psonnera** [reply]
I have little experience with MM3 but start to see new issues as the product diffuses around the world. Whilst it should be fully compatible with the original protocol, I wonder if the new firmware is identical in behaviour. Let me ask them.

**Reply by KimF2020** [reply]
Yeah, i already have contacted them. The connection problem with the Xdrip is back. Same yesterday evenening. Connecting en disconnecting and so on for about 2 hours. And then after 2 hours it's stays connected without doing anything else. Can you think about something what i can do to prevent it?
